includes/class-sensei-course.php

```diff
@@ -100,8 +100,9 @@ public function __construct () {
         add_action('save_post', array( 'Sensei_Course', 'flush_rewrite_rules' ) );
 
 		// Allow course archive to be setup as the home page
-		add_action( 'pre_get_posts', array( $this, 'allow_course_archive_on_front_page' ) );
-
+		if ( (int) get_option( 'page_on_front' ) > 0 ) {
```

Comment:
I worry about this -- Core's test is always simply `if ( get_option( 'page_on_front' ) )` -- 

https://github.com/WordPress/WordPress/blob/9193013158d02ecd51ab8aa52d594aa789ac9836/wp-includes/rewrite.php#L504
https://github.com/WordPress/WordPress/blob/25e66e4f1e5ab91235c2de2fe7d218d0a2f2997d/wp-admin/options-reading.php#L65

With this structured the way it is, if the option got set to something like `'five'` or an unexpected value that can't be parsed to an integer but evaluates to true, we could have a situation where two bits of code are parsing the same value differently.

I'd sooner just use `if ( get_option( 'page_on_front' ) )` barring a really good counter-argument.